src/scripts/generate-test-patient-spreadsheets.sh

```diff
@@ -0,0 +1,186 @@
+  baseUrl=$(jq -r '.baseUrl' "${TEMPLATE_FILE}")
+
+  patientFields=$(jq -r '.patientFields[].name' "${TEMPLATE_FILE}" | tr '\n' ' ')
+  resourceStaticFields=()
+  resourceJqFields=()
+
+  for i in $(seq 0 $(($(jq '.resources | length' "${TEMPLATE_FILE}")-1))); do
+    for staticField in $(jq -r ".resources[${i}].staticFields[].name" "${TEMPLATE_FILE}"); do
+      if [[ ! " ${resourceStaticFields[*]} " =~ [[:space:]]${staticField}[[:space:]] ]]; then
+          resourceStaticFields+=" ${staticField}"
+      fi
+    done
+    for jqField in $(jq -r ".resources[${i}].jqFields[].name" "${TEMPLATE_FILE}"); do
+      if [[ ! " ${resourceJqFields[*]} " =~ [[:space:]]${jqField}[[:space:]] ]]; then
+          resourceJqFields+=" ${jqField}"
+      fi
+    done
+  done
+
+  echo "ICN,${patientFields// /,}Resource${resourceStaticFields// /,}${resourceJqFields// /,}" > "${outputCsv}"
+
+
+  TOKEN=$(new-token)
+
+  # Send the requests and create the spreadsheet rows
+  for patientId in $(jq -r '.testPatients[]' "${TEMPLATE_FILE}"); do
+    request "${baseUrl}/Patient/${patientId}" "${curlResponse}"
+    patientFieldValues="${patientId}"
+    for field in ${patientFields}; do
+      patientFieldValues+=","
+      value=$(jq -r "$(jq -r ".patientFields[] | select(.name==\"${field}\") | .path" "${TEMPLATE_FILE}")" "${curlResponse}")
+      if [ "${value}" != "null" ]; then
+        patientFieldValues+="${value//,/ }"  # Replace commas with spaces
+      fi
+    done
+
+    for i in $(seq 0 $(($(jq '.resources | length' "${TEMPLATE_FILE}")-1))); do
+      resourceType=$(jq -r ".resources[${i}].type" "${TEMPLATE_FILE}")
+      staticFieldValues=""
+      for field in ${resourceStaticFields}; do
+        staticFieldValues+=","
+        value="$(jq -r ".resources[] | select( .type==\"${resourceType}\") | ( .staticFields // [])[] | select(.name==\"${field}\") | .value" "${TEMPLATE_FILE}")"
+        if [ "${value}" != "null" ]; then
+          staticFieldValues+="${value//,/ }"  # Replace commas with spaces
+        fi
+      done
+      createSpreadsheetRowsForResource "${resourceType}" "${outputCsv}" "${curlResponse}" "${patientId}" "${patientFieldValues},${resourceType}${staticFieldValues}" "${resourceJqFields}"
+    done
+  done
+}
+
+createSpreadsheetRowsForResource() {
+  local resourceType="${1}" outputCsv="${2}" curlResponse="${3}" patientId="${4}" rowNonResourceJqFields="${5}" resourceJqFields="${6}" baseUrl jqFieldValues url numRecords
+  baseUrl=$(jq -r '.baseUrl' "${TEMPLATE_FILE}")
+  declare -A jqFieldValues
+
+  url="${baseUrl}/${resourceType}?patient=${patientId}"
+  while [ -n "${url:-}" ]; do
+    jqFieldValues=()
+    request "${url}" "${curlResponse}"
+    unset url
+
+    numRecords=$(jq -r '.entry | length' "${curlResponse}")
+
+    if [ "${numRecords}" -eq 0 ]; then
+      log "INFO" "No ${resourceType} resources found for patient ${patientId}"
+      continue
+    fi
+
+    for jqField in ${resourceJqFields}; do
```

Comment:
So... there is a way to do this in one go... I think it may hurt more than it helps though. I did a similar thing for the fingerprint check script.